### PR TITLE
[DEV-4751] goal templates oa read operations

### PR DESCRIPTION
**Pull request by gaetano-guerriero**

### CODE REVIEW

File: server/athenian/api/align/goals/templates.py [hunk 1]
```diff
@@ -1,37 +1,58 @@
+from __future__ import annotations
+
+from typing import TypedDict
+
 from athenian.api.models.web import JIRAMetricID, PullRequestMetricID, ReleaseMetricID
+from athenian.api.models.web.goal_template import GoalTemplateMetricID
 
-TEMPLATES_COLLECTION = {
+TEMPLATES_COLLECTION: dict[int, _TemplateDefinition] = {
```
Comment (vmarkovtsev) [inline server/athenian/api/align/goals/templates.py:8]:
```suggestion
TEMPLATES_COLLECTION: dict[int, TemplateDefinition] = {
```

Let's make it a fully-featured public class.

File: server/athenian/api/controllers/goal_controller.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    """Retrieve a goal template.
+
+    :param id: Numeric identifier of the goal template.
+    :type id: int
+
+    """
+    try:
+        template_def = TEMPLATES_COLLECTION[id]
+    except KeyError:
+        return ResponseError(NotFoundError("Template %d was not found." % id)).response
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/goal_controller.py:22]: Let's add a big TODO to check the account ID <> template ID access.

**Review by vmarkovtsev — APPROVED** [review]
I don't see anything critical, these are some points to consider

File: server/athenian/api/models/web/goal_template.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        """Gets the metric of this GoalTemplate."""
+        return self._metric
+
+    @metric.setter
+    def metric(self, metric: GoalTemplateMetricID) -> None:
+        """Sets the metric of this GoalTemplate."""
+        if metric is None:
+            raise ValueError("Invalid value for `metric`, must not be `None`")
+
+        self._metric = metric
```
Comment (vmarkovtsev) [inline server/athenian/api/models/web/goal_template.py:70]: We must check if the value is in `JIRAMetricID | PullRequestMetricID | ReleaseMetricID`

File: server/athenian/api/controllers/goal_controller.py (hunk 2 above)
Comment (gaetano-guerriero) [reply server/athenian/api/controllers/goal_controller.py:22]: Next PR will add retrieve from DB, and there we check account ownership

File: server/athenian/api/models/web/goal_template.py (hunk 3 above)
Comment (gaetano-guerriero) [reply server/athenian/api/models/web/goal_template.py:74]: Added.
I just noticed that typing with `base_model_.Enum` is not working as expected, i.e. `metric` is really string.
This is another global issue we must resolve before introducing mypy

Comment (vmarkovtsev) [reply server/athenian/api/models/web/goal_template.py:74]: The string goes directly from aiohttp/connexion, the JSONSchema engine checks enums but doesn't do anything with types, I think. On the other hand, the code is simple now and it's important.

**Pull request merged**